conans/client/remote_manager.py

```diff
@@ -99,12 +100,18 @@ def get_recipe(self, ref, remote):
         self._hook_manager.execute("pre_download_recipe", reference=ref, remote=remote)
 
         ref = self._resolve_latest_ref(ref, remote)
-        layout = self._cache.create_ref_layout(ref)
+        try:
```

Comment:
I think this might be worth moving this in a single call in a ``ClientCache`` class method, (so it is cleaner to optimize it later if necessary, moving down the ``DataCache`` as a single DB query). The ``RemoteManager`` shouldn't be dealing with this Exception, this responsibility doesn't belong here.

Reply:
Moved to `get_or_create_ref_layout` and `get_or_create_pkg_layout` methods in ClientCache.